src/editor/controller/editor_controller.cpp

```diff
@@ -231,13 +232,67 @@ void editor_controller::custom_tods_dialog()
 	}
 
 	tod_manager& manager = *get_current_map_context().get_time_manager();
+	std::vector<time_of_day> prev_schedule = manager.times();
+
+	gui2::dialogs::custom_tod tod_dlg(manager.times(), manager.get_current_time());
+
+	/* Register callback to the dialog so that the map changes can be
+	 * previewed in real time.
+	 */
+	std::function<void(std::vector<time_of_day>)> update_func(
+				std::bind(
+						&editor::editor_controller::update_map_schedule,
+						this,
+						std::placeholders::_1));
+	tod_dlg.register_callback(update_func);
+
+	/* Autogenerate schedule id and name in case the user doesn't give any. */
+	std::int64_t current_millis = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+	std::string sch_id = "custom-"+std::to_string(current_millis);
+	std::string sch_name = "Custom Schedule "+std::to_string(current_millis);
```

Comment:
While we can't add translatable strings at the moment, this ought to be one.

Reply:
I've added a TODO to mark these as translatable strings. (76f27ac3a123cad94a813f97cfd8ff9325ad72f7)